src/windows_registry.rs

```diff
@@ -957,3 +947,73 @@ mod impl_ {
             })
     }
 }
+
+/// Non-Windows Implementation.
+#[cfg(not(windows))]
+mod impl_ {
+    use std::{env, ffi::OsString};
+
+    use super::{TargetArch, MSVC_FAMILY};
+    use crate::Tool;
+
+    /// Finding msbuild.exe tool under unix system is not currently supported.
+    /// Maybe can check it using an environment variable looks like `MSBUILD_BIN`.
+    pub(super) fn find_msbuild(_target: TargetArch<'_>) -> Option<Tool> {
+        None
+    }
+
+    // Finding devenv.exe tool under unix system is not currently supported.
+    // Maybe can check it using an environment variable looks like `DEVENV_BIN`.
+    pub(super) fn find_devenv(_target: TargetArch<'_>) -> Option<Tool> {
+        None
+    }
+
+    /// Attempt to find the tool using environment variables set by vcvars.
+    pub(super) fn find_msvc_environment(tool: &str, _target: TargetArch<'_>) -> Option<Tool> {
+        // Early return if the environment doesn't contain a VC install.
+        let _vc_install_dir = env::var_os("VCINSTALLDIR")?;
```

Comment:
Seem to be another duplicate.

Reply:
We can not do this at a non-Windows platform, so we should remove it, or find out any way to do like this.

Reply:
I mean there are two lines of

```
        let _vc_install_dir = env::var_os("VCINSTALLDIR")?;
```

Reply:
Thanks, these two lines looks really similar that my brain mixed them up

Reply:
> We have to make sure `VCINSTALLDIR ` exists because the build tools are under `VCINSTALLDIR`.

If the build tools are under them, does it make sense to use them to find the tools in the same way it's done below?

Reply:
I think it makes a lot of sense, but I'm not sure if it works or if it's really consistent with real usage.

Reply:
In theory, the VS installation directory structure of the simulation environment (such as Wine) should be consistent with the host environment (Windows host), but I am not sure whether other custom installation behaviors are completely consistent with those installed on the host using vsinstaller.

Reply:
Also, whether we distinguish between non-Windows host environments should not be confused with the assumption that it must be wine here, if a non-Windows host environment is not necessarily by wine, but at least for now.

Reply:
Then again, I think once we have a simulated environment, the entire execution method should not be much different from the real windows environment, so the windows implementation should become available here as well, right? I don't know if the distinction between windows and non-windows is really needed here.

Reply:
I think setting up a new CI for testing cross compilation to Windows on Linux using cross-rs or wine will be helpful.

It would verify that the code you wrote indeed works and gives us confidence in merging in the code, and detect future regression.

Reply:
Unfortunately, since the msvc environment is not redistributable, we don't have a pre-built image to perform CI, and we can't simply do this? May only be made on a non-disclosure and non-redistribution basis.

Reply:
Ahhh yes,that's really annoying.

Since we are only using in the CI, maybe that term doesn't apply to this use case/microsoft won't sue us?

Reply:
It would be great if we can get support from people who are more familiar with them or microsoft employers

Reply:
Although only used in CI, image storage containing msvc tools appears to be against Microsoft policy.

Reply:
That's annoying...